backend/bookings/services/bookingService.js

```diff
@@ -1,23 +1,28 @@
-const { sequelize } = require("@utopia-airlines-wss/common/db");
-const { Booking } = require("@utopia-airlines-wss/common/models");
-const {  NotFoundError,  handleMutationError   } = require("@utopia-airlines-wss/common/errors");
+const {sequelize} = require("@utopia-airlines-wss/common/db");
+const {Booking} = require("@utopia-airlines-wss/common/models");
+const {NotFoundError, handleMutationError} = require("@utopia-airlines-wss/common/errors");
 
 const findBookingById = async (id, options) => {
   const booking = await Booking.findByPk(id, options);
   if (!booking) throw new NotFoundError("cannot find booking");
   return booking;
 };
 
-const replacer = (key, value) =>  key === "passengers"
+const replacer = (key, value) => key === "passengers"
   ? value.map((passenger) => passenger.toJSON("full"))
   : value;
 
-
 const bookingService = {
-  async findAllBookings({ isActive = true } = {}) {
-    const where = { isActive };
-    const bookings = await Booking.findAll({
+  async findAllBookings(
+    {isActive = true, offset = 0, limit = 10} = {}) {
+    const where = {isActive};
+    limit = limit ? Math.min(+limit, 40) : 10;
+    offset = offset ? offset * limit : 0;
```

Comment:
It should really be possible to set an arbitrary offset that can be partway through a "page".
So something like `offset = offset ?? 0`

Reply:
Hmmm, The approach that takes offset and limit into account looks more cleaner and elegant to me.

Reply:
It is more about giving the front end as much control as possible so we don't have to change the backed later. If we make it so that offset and limit are linked in the implementation on the backend then the front end really only has control over the limit. There isn't a need to force the front end to request a section of the data that is a multiple of the limit. Let the front end handle multiplying page by the limit.

Reply:
I'm not sure about that. I think having some control in the backend is a better approach. And front-end has enough control to pass a combination of offset and limit.

Reply:
The client has a better understanding of how it is using the data. It is true that for how were are implementing it currently, it doesn't make much of a difference. That being said, it does drastically limit how our API can be consumed for no real benefit. If someone is trusted enough to make a request get any booking they should be trusted to choose what sub section they want.

Reply:
To me it is more clear and elegant that when you run `?offset=2&limit=10` see bookings from 20..30 as oppose to 2...12. We don't want to have large numbers get passed to our api calls as offset. (e.g, `?offset=4143243&limit=10`). You are right on the fact that currently, it doesn't make much of a difference. But I don't think at any point in future our design ever requires that. In other words I can't see how this going to bring any value to our design.